src/unification.cpp

```diff
@@ -0,0 +1,82 @@
+#include "unification.hpp"
+
+#include <cassert>
+
+namespace Unification {
+
+bool Core::occurs(int v, int i){
+	assert(node_header[v].tag == Tag::Var);
+	assert(node_header[v].data_idx == v);
+
+	i = find(i);
+
+	if (i == v)
+		return true;
+
+	assert(node_header[i].tag == Tag::Term);
```

Comment:
did I write this? hahah. I don't think that assertion is right.

Reply:
`i` can still be a Var that's not equal to `v`, can't it?

Reply:
It shouldn't, the way we use it, in any case we should return false instead of asserting